File: app/src/main/java/net/tlalka/android/fiszki/activities/BaseSetupActivity.java
package net.tlalka.android.fiszki.activities;

import android.view.Menu;
import android.view.MenuItem;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.elements.SetupElement;

public abstract class BaseSetupActivity extends AbstractActivity {

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return this.createMenu(R.menu.setup_menu, menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return SetupElement.triggerAction(this, item.getItemId()) || super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/view/activities/BaseSetupActivity.kt
package net.tlalka.fiszki.view.activities

import android.view.Menu
import android.view.MenuItem
import net.tlalka.fiszki.R
import net.tlalka.fiszki.view.elements.SetupElement

abstract class BaseSetupActivity : AbstractActivity() {

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return createMenu(R.menu.setup_menu, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return SetupElement.triggerAction(this, item.itemId) || super.onOptionsItemSelected(item)
    }
}
